src/lib/secrets/privkey_error.ml

```diff
@@ -0,0 +1,39 @@
+open Core
+
+exception Corrupted_privkey of Error.t
+
+exception Incorrect_password_or_corrupted_privkey
+
+exception Cannot_open_file of string
+
+exception Parent_directory_does_not_exist of string
+
+type t =
+  | Corrupted_privkey of Error.t
+  | Incorrect_password_or_corrupted_privkey
+  | Cannot_open_file of string
+  | Parent_directory_does_not_exist of string
```

Comment:
Can you make the exception:

```
exception Privkey_exn of t
```

and that way we don't duplicate the code too much here